### PR TITLE
Fixes #30365: Ignore symlinks for puppet executable

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ekohl** [comment]
Looks like the tests fail on this

**Comment by ehelms** [comment]
Yea, due to paths existing on local system I have to run these tests in the isolated Jenkins env >.<

**Comment by ehelms** [comment]
Tests passing now

**Comment by ehelms** [comment]
Related fix needed: https://github.com/theforeman/kafo_parsers/pull/40

**Comment by ehelms** [comment]
@ekohl did you want to look at this for pulling into a release?

File: lib/kafo/puppet_command.rb [hunk 1]
```diff
@@ -38,8 +38,11 @@ def self.search_puppet_path(bin_name)
       # Find the location of the puppet executable and use that to
       # determine the path of all executables
       bin_path = (::ENV['PATH'].split(File::PATH_SEPARATOR) + ['/opt/puppetlabs/bin']).find do |path|
```
Comment (ekohl) [inline lib/kafo/puppet_command.rb:40]: I still have doubts about this line. In production we actually want to prefer the last value over `$PATH` but in dev we don't. However, I think it's OK. The user with a non-trivial wrapper script could fix it and that's probably ok.

File: test/test_helper.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@
 require 'minitest/spec'
 require 'minitest/mock'
 require "minitest/reporters"
+require 'mocha/minitest'
```
Comment (ekohl) [inline test/test_helper.rb:9]: Why mocha? We now have 2 ways to stub/mock.

Comment (ehelms) [reply test/test_helper.rb:9]: Because minitest stubbing is insufficient to handle our use case.

**Comment by ehelms** [comment]
@ekohl Anything else you need here to complete review?

File: test/test_helper.rb (hunk 2 above)
Comment (ekohl) [reply test/test_helper.rb:9]: I'd prefer to get 4d26925692cda13fb52f0382686d0be72be6655c in to avoid the use of mocha.

**Comment by ekohl** [comment]
Looks like CI is still running, but :+1: other than that.

**Pull request merged**